spec-0008/index.md

```diff
@@ -0,0 +1,228 @@
+
+### Document the release process
+
+The release process should be clearly and fully documented in the developer documentation and describe each step to make a release and the permissions required to do so.
+It is recommended that this is a dedicated page in the developer section of the documentation website, though providing instructions in a `RELEASING.md` in the top level of the repository is also a common approach.
+
+### Hardening workflow environment permissions
+
+Workflows that publish release artifacts should have _run triggers_ that require intentional actions by maintainers (e.g., `workflow_dispatch` in GitHub Actions) and require multiple maintainers to approve the workflow to run (c.f. "Use GitHub Actions environments" section below).
+This is to safeguard the project from any one maintainer having the ability to commit to the default branch and make a release directly.
+
+It is also strongly recommended that the repository requires [signed commits](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) so that each release corresponds to a verified commit.
+The branch from which the release is made should also be protected.
+
+#### Restrict permissions in CI runners to the minimum required
+
+To restrict the attack surface area of arbitrary code execution in CI runners, the default permissions the runners have should be restricted to the minimum possible (read access). In the GitHub Action workflow, this is accomplished by defining the following workflow global permissions block before any jobs are defined.
+
+```yaml
+permissions:
+  contents: read
+```
+
+Elevating permissions beyond this should be done at the job level by redefining the permissions block in the job.
+
+#### Restrict permitted actions in workflows
+
+GitHub allows restricting the actions that workflows can use via the repository actions permissions settings at `https://github.com/$ORG/$PROJECT/settings/actions`.
+A reasonable default is to select
+
+> Allow $ORG, and select non-$ORG, actions and reusable workflows
+
+and the suboptions:
+
+* Allow actions created by GitHub
+* Allow specified actions and reusable workflows
+
+Consult [Managing GitHub Actions permissions for your repository](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#managing-github-actions-permissions-for-your-repository) for more details.
+
+#### Use GitHub Actions environments
+
+Use a [GitHub Actions environment](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment)
+
+```yaml
+environment:
+  name: publish-package
+```
+
+and enforce additional review by at least one other maintainer to run a GitHub Actions workflow that publishes to PyPI.
```

Comment:
It is mentioned above, yes, but we are intentionally doing so (above we said

> (c.f. "Use GitHub Actions environments" section below)

)

so that we get the idea forward and then expand on it here.